OpenQA/Isotovideo/CommandHandler.pm

```diff
@@ -366,4 +376,51 @@ sub _handle_command_send_clients {
     $self->_respond_ok();
 }
 
+sub status_finished {
+    update_test_status(sub {
+        my ($data) = @_;
+        $data->{running} = 0;
+        $data->{finished} = 1;
+        $data->{test_execution_paused} = 0;
+        $data->{current_test} = '';
+    });
+}
+
+sub status_set_test {
+    my ($status) = @_;
+
+    update_test_status(sub {
+        my ($data) = @_;
+        delete $data->{initial};
+        $data->{running} = 1;
+        $data->{test_execution_paused} = $status->{test_execution_paused};
+        if ($status->{current_test}) {
+            # dont't overwrite the last running test for information
+            $data->{current_test} = $status->{current_test};
+            $data->{tests}->{ $data->{current_test} } = 1;
+        }
+    });
+
+}
+
+sub update_test_status {
+    my ($code) = @_;
+
+    my $coder = Cpanel::JSON::XS->new->pretty->canonical;
+
+    open my $fh, '+<', AUTOINST_STATUSFILE or die $!;
+    flock $fh, LOCK_EX;
+    my $json = do { local $/; <$fh> };
+    my $data = $coder->decode($json);
```

Comment:
I wasn't sure if that would be sufficient because the module is handling events, so it works asynchronously.

Maybe it would be sufficient, but then we couldn't let other processes modify the file (if we ever need that feature)

Reply:
I wouldn't let other process modify the file. In my opinion it would be cleaner to let them always query isotovideo to modify the file.